CSharp/Library/Microsoft.Bot.Connector.Shared/ConnectorClientEx.cs

```diff
@@ -49,25 +49,44 @@ public ConnectorClient(Uri baseUri, MicrosoftAppCredentials credentials, HttpCli
         {
             this.Credentials = credentials;
         }
-        
+
         private static DelegatingHandler[] AddJwtTokenRefresher(DelegatingHandler[] srcHandlers, MicrosoftAppCredentials credentials)
         {
             var handlers = new List<DelegatingHandler>(srcHandlers);
             handlers.Add(new JwtTokenRefresher(credentials));
             return handlers.ToArray();
         }
-        
-        // client defaults to sending the expect: continue header, which isn't very efficient, 
+
+        private HttpClient instanceClient;
+        protected static HttpClient g_httpClient = null;
+        protected static object syncObj = new object();
+
         partial void CustomInitialize()
         {
-            AddUserAgent(this);
-            HttpClient.DefaultRequestHeaders.ExpectContinue = false;
+            if (g_httpClient == null)
+            {
+                lock (syncObj)
+                {
+                    if (g_httpClient == null)
+                    {
+                        g_httpClient = new HttpClient();
+                        g_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Microsoft-BotFramework", "3.1"));
+                        g_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue($"(BotBuilder .Net/{GetClientVersion(this)})"));
+                        g_httpClient.DefaultRequestHeaders.ExpectContinue = false;
+                    }
+                }
+            }
+
+            // use global singleton for perf reasons
+            this.instanceClient = this.HttpClient;
+            this.HttpClient = g_httpClient;

```

Comment:
This is a big hack:
1. Creates a static global, in a lock. Clear 'nuf.f
2. Preserve the previous httpClient. 
3. Swap in the global one. 
4. Later, in the Dispose, swap the instance one back in to be disposed. 

Do this weird behavior w/o comments in the code as to whyat's going on and why. 

There's got to be something cleaner.